### DISCUSSION TITLE
Do snippets get injected into the page before they are parsed by the theme?

### DISCUSSION THREAD

**Discussion by Andre601** [facelessuser/pymdown-extensions#2122] [Q&A] [answered]
Was wondering of the execution order of the snippet extension when used in combination with the Material for MkDocs theme.

The reason I ask is, that I would like to have SVG icons included within some snippets using the `{% include "..." %}` tag of jinja2 and was wondering if snippets puts the content in before Material for MkDocs parses over it and replaces the include tag with the SVG icon.

...I hope what I ask about makes sense...

**Comment by facelessuser** [comment]
Snippets are applied at the very beginning of the Markdown processing step. Does MkDocs process `{% include "..." %}` in Markdown files normally? I honestly don't know.

**Reply by Andre601** [reply]
I don't know either...

Tho, perhaps the default `:name:` pattern of the Emoji extension will work... given that it runs after snippets that is...

**Reply by Andre601** [reply, marked as answer]
I actually can't use the `:name:` pattern because I use HTML... So I simply directly included the full SVG itself in a span object to use the twemoji class...

(Also, I use HTML to not have permalinks auto-added to headers)

**Reply by facelessuser** [reply]
You can use `:name:` in HTML, but you'd need to use `md_in_html` extension. But if you have something working, that's all that matters.
